test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -231,21 +231,46 @@ void ProcessExport(Batch<Metric> batch)
         [Fact]
         public void MultithreadedLongCounterTest()
         {
-            var metricItems = new List<Metric>();
-            var metricExporter = new TestExporter<Metric>(ProcessExport);
+            this.MultithreadedCounterTest(deltaLongValueUpdatedByEachCall);
+        }
 
-            void ProcessExport(Batch<Metric> batch)
+        [Fact]
+        public void MultithreadedDoubleCounterTest()
+        {
+            this.MultithreadedCounterTest(deltaDoubleValueUpdatedByEachCall);
+        }
+
+        [Fact]
+        public void MultithreadedLongHistogramTest()
+        {
+            this.MultithreadedHistogramTest(deltaLongValueUpdatedByEachCall);
+        }
+
+        [Fact]
+        public void MultithreadedDoubleHistogramTest()
+        {
+            this.MultithreadedHistogramTest(deltaDoubleValueUpdatedByEachCall);
+        }
+
+        [Fact]
+        public void MultithreadedGaugeTest()
+        {
+            Assert.True(true);
+        }
+
+        private void MultithreadedCounterTest<T>(T deltaValueUpdatedByEachCall)
+            where T : struct, IComparable
+        {
+            var metricItems = new List<Metric>();
```

Comment:
Could `metricItems` be replaced with a `T sum` and in the `foreach` loop below replace the statement with looping through the `MetricPoint`s  and calling `sum += metricPoint.DoubleValue` or `sum += metricPoint.LongValue`. Making this `List<Metric>` and making the below `GetLongSum` and `GetDoubleSum` function unnecessary?